lib/utils.js

```diff
@@ -367,52 +368,56 @@ function parseSettings(urlStr) {
 }
 
 /**
- * Merge model settings
+ * Objects deep merge
  *
- * Folked from https://github.com/nrf110/deepmerge/blob/master/index.js
+ * Forked from https://github.com/nrf110/deepmerge/blob/master/index.js
  *
- * The original function tries to merge array items if they are objects
+ * The original function tries to merge array items if they are objects, this
+ * was changed to always push new items in arrays, independently of their type.
  *
- * @param {Object} target The target settings object
- * @param {Object} src The source settings object
- * @returns {Object} The merged settings object
+ * NOTE: The function operates as a deep clone when called with a single object
+* argument.
+ *
+ * @param {Object} base The base object
+ * @param {Object} obj The object to merge with base object
+ * @returns {Object} The merged object
  */
-function mergeSettings(target, src) {
-  var array = Array.isArray(src);
+function deepMerge(base, obj) {
```

Comment:
`target -> base` make sense to me. I find the name `obj` too generic, I had to read jsdoc to understand what's that argument. Can we use a different name, e.g. something like `extensions` or `extra`?